design_drafts/ros2_control_documentation.rst

```diff
@@ -0,0 +1,273 @@
+   
+Repositories
+============
+The ros2_control framework consist of the following repositories:
+
+ros2_control
+  The `ros2_control`_ repository implements the main interfaces and components of the framework mentioned in the previous sections.
+  
+ros2_controllers
+  The `ros2_controllers`_ repository implements widely used controllers, e.g., forward controller, joint trajectory controller, differential drive controller, etc.
+  
+ros2_control_demos
+  The `ros2_control_demos`_ repository provides examples of using the framework and templates for a smooth start with it.
+
+Differences to ros_control (ROS1)
+=================================
+
+Hardware Structures - classes
+-----------------------------
+
+The ros_control uses the ``RobotHW`` class as a rigid structure to handle any hardware.
+This makes it impossible to extend the existing robot with additional hardware, like sensors, actuators, and tools, without coding.
+
+The ros2_control defines three different types of hardware ``Actuator``, ``Sensor`` and ``System``.
+Using a combination (composition) of those basic components, any physical robotic cell (robot and its surrounding) can be described.
+This also means that multi-robot, robot-sensor, robot-gripper combinations are supported out of the box.
+Section `Hardware Components <#hardware-components>`__ describes this in detail.
+
+Hardware Interfaces
+-------------------
+
+The ros_control allows only three types of interfaces (joints), i.e., ``position``, ``velocity``, and ``effort``. The ``RobotHW`` class makes it very hard to use any other data to control the robot.
+
+The ros2_control does not mandate a fixed set of interface types, but they are defined as strings in `hardware's description <#hardware-description-in-urdf>`__.
+To ensure compatibility of standard controllers, standard interfaces are defined as constants in `hardware_interface package <https://github.com/ros-controls/ros2_control/blob/master/hardware_interface/include/hardware_interface/types/hardware_interface_type_values.hpp>`__.
+
+Controller's Access to Hardware
+-------------------------------
+
+In ros_control, the controllers had direct access to the ``RobotHW`` class requesting access to its interfaces (joints).
+The hardware itself then took care of registered interfaces and resource conflicts.
+
+In ros2_control, ``ResourceManager`` takes care of the state of available interfaces in the framework and enables controllers to access the hardware.
+Also, the controllers do not have direct access to hardware anymore, but they register their interfaces to the `ControllerManager`.
+
+Migration Guide to ros2_control
+===============================
+
+RobotHardware to Components
+---------------------------
+#. Forget your implementation or the ``RobotHW`` interface. This is not used anymore. (Do not delete it, you can still extract some code.)
```

Comment:
This is a little harsh :D I'd say that typically, one's `RobotHW` can be migrated directly to a `SystemInterface`, if one wants more granularity, they can branch some parts out into `SensorInterface` or `ActuatorInterface`